internal/output/table.go

```diff
@@ -118,9 +118,11 @@ func printSummaryResult(result Result, outputWriter io.Writer, terminalWidth int
 	}
 
 	for _, ecosystem := range result.Ecosystems {
-		fmt.Fprintln(outputWriter, ecosystem.Name)
-
 		for _, source := range ecosystem.Sources {
+			if source.PackageTypeCount.Regular == 0 {
+				break
+			}
+			fmt.Fprintln(outputWriter, ecosystem.Name)
```

Comment:
let's keep this statement outside the for loop because an ecosystem can have multiple sources. If you want to avoid printing the ecosystem name when all sources lack regular package types, consider creating a helper function. This function could check for if regular packages exist in any source and then use to determine whether to print the ecosystem name.